dropwizard-rabbitmq/src/main/java/smartthings/dw/rabbitmq/connection/ChannelPoolFactory.java

```diff
@@ -0,0 +1,41 @@
+package smartthings.dw.rabbitmq.connection;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.PooledObjectFactory;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+
+public class ChannelPoolFactory implements PooledObjectFactory<Channel> {
+
+    private final Connection connection;
+
+    public ChannelPoolFactory(Connection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public PooledObject<Channel> makeObject() throws Exception {
+        return new DefaultPooledObject<>(connection.createChannel());
+    }
+
+    @Override
+    public void destroyObject(PooledObject<Channel> p) throws Exception {
```

Comment:
I would prefer `pool` over `p`, nitpick

Reply:
Updated it, all of those were just auto-generated by IntelliJ 🥇